python_modules/dagster/dagster/_core/definitions/sensor_definition.py

```diff
@@ -1408,11 +1409,29 @@ def _run_requests_with_base_asset_jobs(
         else:
             asset_keys = outer_asset_selection.resolve(asset_graph)
 
+        if run_request.asset_check_keys is not None:
+            asset_check_keys = run_request.asset_check_keys
+
+            unexpected_asset_check_keys = (
+                AssetCheckKeysSelection(selected_asset_check_keys=asset_check_keys)
+                - outer_asset_selection
+            ).resolve_checks(asset_graph)
+            if unexpected_asset_check_keys:
+                raise DagsterInvalidSubsetError(
+                    "RunRequest includes asset check keys that are not part of sensor's asset_selection:"
+                    f" {unexpected_asset_check_keys}"
+                )
```

Comment:
so this is technically a breaking change right? Before asset checks did not break this.

Probably safer to warn for now?

Reply:
Yeah it is breaking, though I would argue it's a bugfix. But I see how it might cause a user a problem. I updated to emit a deprecation warning (for 1.9) instead.